### PR TITLE
feat(headless): document DictionaryFieldContext

### PR DESCRIPTION
**Pull request by samisayegh**
https://coveord.atlassian.net/browse/KIT-354

### CODE REVIEW

File: packages/headless/src/features/dictionary-field-context/dictionary-field-context-actions-loader.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import {DictionaryFieldContextPayload} from './dictionary-field-context-state';
+
+export {AddDictionaryFieldContextActionCreatorPayload};
+
+/**
+ * The dictionary field context action creators.
+ */
+export interface DictionaryFieldContextActionCreators {
+  /**
+   * Adds a new dictionary field context value.
```
Comment (ThibodeauJF) [inline packages/headless/src/features/dictionary-field-context/dictionary-field-context-actions-loader.ts:19]: Is it equivalent to: "Adds a dictionary field context field-key pair." ?

**Review by ThibodeauJF — APPROVED** [review]

**Review by btaillon — APPROVED** [review]

File: packages/samples/headless-react/src/components/dictionary-field-context/dictionary-field-context.fn.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import {buildDictionaryFieldContext} from '@coveo/headless';
+import {useContext} from 'react';
+import {AppContext} from '../../context/engine';
+
+export function DictionaryFieldContext() {
+  const {engine} = useContext(AppContext);
+  const ctx = buildDictionaryFieldContext(engine!);
+
+  ctx.set({price: 'usd', city: 'boston'});
```
Comment (olamothe) [inline packages/samples/headless-react/src/components/dictionary-field-context/dictionary-field-context.fn.ts:9]: I don't think that's a particularly good "real life" example/sample. 

I would poke someone from commerce team for this.

**Review by olamothe — APPROVED** [review]

File: packages/headless/src/features/dictionary-field-context/dictionary-field-context-actions-loader.ts (hunk 1 above)
Comment (samisayegh) [reply packages/headless/src/features/dictionary-field-context/dictionary-field-context-actions-loader.ts:19]: Yep, adjusted, thanks

File: packages/samples/headless-react/src/components/dictionary-field-context/dictionary-field-context.fn.ts (hunk 2 above)
Comment (samisayegh) [reply packages/samples/headless-react/src/components/dictionary-field-context/dictionary-field-context.fn.ts:9]: @francoislg @dlafreniere what would you suggest?

Comment (dlafreniere) [reply packages/samples/headless-react/src/components/dictionary-field-context/dictionary-field-context.fn.ts:9]:
```
{ "price_currency": "fr", "price_discounted": "fr" }
```
is from a real client, but the example isn't so bad

Comment (samisayegh) [reply packages/samples/headless-react/src/components/dictionary-field-context/dictionary-field-context.fn.ts:9]: Thanks Devin,

**Pull request merged**